library/data/ir_actions_server.xml

```diff
@@ -0,0 +1,48 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="stock_rule_set_push_domain_server_action" model="ir.actions.server">
+        <field name="name">Execute Code</field>
+        <field name="model_id" ref="stock.model_stock_rule"/>
+        <field name="state">code</field>
+        <field name="code"><![CDATA[
+for record in records:
+    if record.x_studio_standard_rule or not record.x_studio_pos_product_category_ids:
+        record['push_domain'] = ''
+        break;
+    domain = [("product_id.pos_categ_ids", "in", record.x_studio_pos_product_category_ids.ids)]
+    record['push_domain'] = domain
+]]></field>
+    </record>
+    <record id="action_update_automatic_activation_on_product" model="ir.actions.server">
+        <field name="code"><![CDATA[
+for record in records:
+    if record.product_selectable:
+        record.write({'x_studio_automatic_activation_on_product': False})
+    else:
+        record.write({'x_studio_automatic_activation_on_product': True})
```

Comment:
```suggestion
records.filtered(lambda r: r.product_selectable).write({
    'x_studio_automatic_activation_on_product': False
})
records.filtered(lambda r: not r.product_selectable).write({
    'x_studio_automatic_activation_on_product': True
})
```
This will call write method 2 times instead of N.